### PR TITLE
Add check to not use deprecated Cypher syntax when Neo4j version is >= 5.23.0

### PR DESCRIPTION
**Pull request by willtai**
# Description
- Add check to not use deprecated Cypher syntax when Neo4j version is >= 5.23.0
- Removes warning supression

## Type of Change
- [x] New feature
- [ ] Bug fix
- [ ] Breaking change
- [ ] Documentation update
- [ ] Project configuration change

## Complexity

Complexity: Low

## How Has This Been Tested?
- [x] Unit tests
- [x] E2E tests
- [x] Manual tests

# Checklist

The following requirements should have been met (depending on the changes in the branch):

- [ ] Documentation has been updated
- [x] Unit tests have been updated
- [ ] E2E tests have been updated
- [ ] Examples have been updated
- [ ] New files have copyright header
- [x] CLA (https://neo4j.com/developer/cla/) has been signed
- [x] CHANGELOG.md updated if appropriate

### CODE REVIEW

**Comment by willtai** [comment]
@CodiumAI-Agent /update_changelog

**Review by stellasia — COMMENTED** [review]
Can you also take a look at the hybrid retriever query please? We have the same deprecation warning.

**Review by stellasia — COMMENTED** [review]
Code LGTM.

Is there an E2E test where we could check that the log is not printed (to prevent any regression)?

File: tests/e2e/docker-compose.yml [hunk 1]
```diff
@@ -26,7 +26,7 @@ services:
     environment:
       ENABLE_CUDA: "0"
   neo4j:
-    image: neo4j:5-enterprise
+    image: neo4j:5.24-enterprise
```
Comment (willtai) [inline tests/e2e/docker-compose.yml:29]: I had to fix this to ensure we get the latest Cypher syntax changes

File: tests/e2e/utils.py [hunk 2]
```diff
@@ -17,6 +17,7 @@
 import hashlib
 import json
 import os.path
+from contextlib import contextmanager
```
Comment (stellasia) [inline tests/e2e/utils.py:20]: Why do we need this import?

File: tests/e2e/docker-compose.yml (hunk 1 above)
Comment (stellasia) [reply tests/e2e/docker-compose.yml:29]: Do you know which version was pulled if not the latest? The tag has been updated recently, could that have an impact?

Comment (willtai) [reply tests/e2e/docker-compose.yml:29]: I could not reproduce the warnings with `5-enterprise`, so it must be 5.22 or below. After adding this I can reproduce the warnings

Comment (stellasia) [reply tests/e2e/docker-compose.yml:29]: Which version are we using in CI?

Comment (stellasia) [reply tests/e2e/docker-compose.yml:29]: (sorry I'm nitpicking on this, mainly because I'm not sure about the implications of pinning the version in e2e)

**Review by stellasia — APPROVED** [review]
📈

**Pull request merged**